In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("usarrests.csv")

In [3]:
df.head()

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91


In [4]:
df.columns

Index(['Unnamed: 0', 'Murder', 'Assault', 'UrbanPop'], dtype='str')

In [5]:
df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [6]:
df[df["UrbanPop"] > 100]

,Unnamed: 0,Murder,Assault,UrbanPop
14,Iowa,2.2,56.0,570


It is Imposible percenatsge (the urbanpop represent 570% which represnt imposible result which eceed the logic )the maximuim can't be more 100% , My acction will be check the source it should be a data entry mistake (it might be 57%) which make more since

In [7]:
df[df["Assault"] > 500]

,Unnamed: 0,Murder,Assault,UrbanPop
39,South Carolina,14.4,879.0,48


The south carolina Assult value is extermly high which represent 879 3 times the value of the close result for any other state .however, is not imposible.double check the source value one more time to confirm if it correct or not .

In [8]:
df[df["Assault"].isna()]

,Unnamed: 0,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60


In [9]:
# Filter the dataframe for rows where UrbanPop is 6
suspicious_row = df[df['UrbanPop'] == 6]
print(suspicious_row)

   Unnamed: 0  Murder  Assault  UrbanPop
31   New York    11.1    254.0         6


the 6 value also is the min represent 6 which is very locw value for the urbanpop ,it is not Imposible however , it is extermly value comparing to the other state , my Action is to check the source again to confirm is it correct value or data entry mistake.